driver-core/src/main/com/mongodb/connection/netty/NettyStream.java

```diff
@@ -152,13 +198,24 @@ public void initChannel(final SocketChannel ch) {
                             enableHostNameVerification(sslParameters);
                         }
                         engine.setSSLParameters(sslParameters);
-                        ch.pipeline().addFirst("ssl", new SslHandler(engine, false));
+                        pipeline.addFirst("ssl", new SslHandler(engine, false));
                     }
+
                     int readTimeout = settings.getReadTimeout(MILLISECONDS);
+                    final long readTimeoutMillis;
```

Comment:
Default to NO_SCHEDULE_TIME and then you can remove the else block.

Reply:
Thanks! This, combined with https://github.com/mongodb/mongo-java-driver/pull/636#discussion_r559500203 and https://github.com/mongodb/mongo-java-driver/pull/636#discussion_r559502153 makes the changes look so much cooler 😎